src/io/github/sspanak/tt9/ime/TraditionalT9.java

```diff
@@ -271,6 +274,11 @@ protected boolean onRight(boolean repeat) {
 	protected boolean onNumber(int key, boolean hold, int repeat) {
 		String currentWord = getComposingText();
 
+		if(mInputMode.isABC()) {
```

Comment:
Perhaps, this would be a better alternative?

```suggestion
		if (delayHandler.hasCallbacks(delayAction)) {
```